streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -703,6 +703,16 @@ void runOnce() {
             // we record the ratio out of the while loop so that the accumulated latency spans over
             // multiple iterations with reasonably large max.num.records and hence is less vulnerable to outliers
             taskManager.recordTaskProcessRatio(totalProcessLatency, now);
+
+            // Only log this at INFO when we actually processed some records to avoid flooding INFO-level logs when
+            // there are no records on the input topics
+            if (totalProcessed > 0) {
+                log.info("Processed {} records for tasks {}, polling for more records now",
+                         totalProcessed, taskManager.activeTaskIds());
+            } else {
+                log.debug("Processed {} records for tasks {}, polling for more records now",
```

Comment:
This line looks about the same as L665. I'm wondering if we really need the "else" in this case.

Reply:
Fair enough